src/main/java/in/zapr/druid/druidry/client/DruidJerseyClient.java

```diff
@@ -0,0 +1,144 @@
+        this.jerseyConfig = jerseyConfig;
+
+        log.debug("Will query to {}", druidUrl);
+        this.druidConfiguration = druidConfiguration;
+    }
+
+    @Override
+    public void connect() throws ConnectionException {
+
+        try {
+            if (jerseyConfig == null) {
+
+                HttpClientConnectionManager connectionManager = createConnectionManager();
+                this.jerseyConfig = new ClientConfig();
+                this.jerseyConfig.property(ApacheClientProperties.CONNECTION_MANAGER, connectionManager);
+                this.jerseyConfig.connectorProvider(new ApacheConnectorProvider());
+            }
+
+            this.client = ClientBuilder.newClient(this.jerseyConfig);
+            this.queryWebTarget = this.client.target(this.druidUrl);
+
+        } catch (Exception e) {
+            throw new ConnectionException(e);
+        }
+    }
+
+    @Override
+    public void close() throws ConnectionException {
+        try {
+            if (this.client == null) {
+                return;
+            }
+
+            this.client.close();
+        } catch (Exception e) {
+            throw new ConnectionException(e);
+        }
+    }
+
+    @Override
+    public String query(DruidQuery druidQuery) throws QueryException {
+
+        try (Response response = this.queryWebTarget
+                .request(MediaType.APPLICATION_JSON)
+                .post(Entity.entity(druidQuery, MediaType.APPLICATION_JSON))) {
+
+            if (response.getStatus() == Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()) {
+                handleInternalServerResponse(response);
+            }
+
+            return response.readEntity(String.class);
+
+        } catch (QueryException e) {
+            log.error("Exception while querying {}", e);
+            throw e;
+        } catch (Exception e) {
+            log.error("Exception while querying {}", e);
+            throw new QueryException(e);
+        }
+    }
+
+    @Override
+    public <T> List<T> query(DruidQuery druidQuery, Class<T> className) throws QueryException {
+        try (Response response = this.queryWebTarget
+                .request(MediaType.APPLICATION_JSON)
+                .post(Entity.entity(druidQuery, MediaType.APPLICATION_JSON))) {
+
+            if (response.getStatus() == Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()) {
+                handleInternalServerResponse(response);
+            }
+
+            return response.readEntity(new GenericType<List<T>>() {
+            });
+        } catch (QueryException e) {
+            log.error("Exception while querying {}", e);
+            throw e;
+        } catch (Exception e) {
+            log.error("Exception while querying {}", e);
+            throw new QueryException(e);
+        }
+    }
+
+    private DruidError handleInternalServerResponse(Response response) throws Exception {
+        DruidError error = response.readEntity(DruidError.class);
```

Comment:
Will readEntity() works as DruidError has only getter methods?

Reply:
Code is checked by mocked server. It will work.